app/src/main/java/pl/edu/pjwstk/s999844/shoppinglist/settings/Settings.kt

```diff
@@ -41,6 +41,9 @@ class Settings(context: Context) {
 
 		private const val ORDER_NAME = "listOrder"
 		private val ORDER_DEFAULT: Order = Order.Unordered
+
+		private const val ORDER_ZERO_ITEMS_LAST_NAME = "orderZeroItemsLast"
+		private val ORDER_ZERO_ITEMS_LAST_DEFAULT = true
```

Comment:
Can be `const`
```suggestion
		private const val ORDER_ZERO_ITEMS_LAST_DEFAULT = true
```

Reply:
haa thanks, that's safer.
FWIW I'm not used to kotlin yet (but why enum can't be `const`?), it's actually the first time in writing some.

Reply:
> why enum can't be `const`?

I'm not a kotlin-pro either, but from what I understood enums in kotlin are just a special kind of class and not a numeric type. Since kotlin compiles down to code understandable by the JVM and it's not possible for class instances to be `const` there, enums can't be in kotlin. You may also notice that in the declaration at the bottom of the file in our case they even implement an interface (which I use to automatically handle both color and order dropdowns with only one implementation):

```kotlin
enum class AccentColor(val value: Int, val styleResourceId: Int, private val description: Int) : DescriptiveSetting {
```

Reply:
right, the JVM is kinda old in this regard, as the enum are, as you said, classes and not discrete values of a common type. it might be enforcable at the kotlin compiler level but would be weird to interface with classic java enum. I think that Scala does it better, having sealed classes to work around it.
anyway, thanks for the explanation!